### DISCUSSION TITLE
Dynamic README shield behavior and Package Author alerting

### DISCUSSION THREAD

**Discussion by bdrelling** [SwiftPackageIndex/SwiftPackageIndex-Server#1989] [Q&A] [answered]
I leverage the Swift Package Index [README shields](https://swiftpackageindex.com/bdrelling/InstrumentKit/information-for-package-maintainers) for my packages. These shields "show always-up-to-date compatibility status for your package."

The problem is that SPI can fail, either due to some unforeseen bug in the package _or_ it's due to SPI's CI workflow, and in the meantime SPI has removed a supported platform from my README.

No build system can be 100% accurate, any CI/CD workflow is prone to flakiness. I've seen SPI fail on my packages very infrequently by error, but it has happened, and this is unavoidable.

Is there any way to receive alerting from SPI when one of my packages fails to build, or specifically when the versions/platforms _lose_ compatibility, and there is nothing obvious reflected in the code (eg. `Package.platforms` updated or `swift-tools-version` incremented) that would indicate the change was intentional?

Without any alerting or monitoring for package authors, the README on my packages could potentially update and reflect an inaccurate picture of my package, for one reason or another, and I'll be blissfully unaware until/unless I happen to be monitoring SPI or my README on my own like a hawk.

**Comment by daveverwer** [comment, marked as answer]
This has come up a few times and it's not something we're planning on working on right in the short term. The best way to be notified of compatibility failures as a package author is to set up a few GH Action jobs on the package's repository. You'll get immediate notification of any issues.

It's probably not what you want to hear, but it's the best solution for the foreseeable future.

If you wanted to set up monitoring of the badge to know if/when *we* get it wrong, then monitoring this endpoint is your best bet:

```
https://swiftpackageindex.com/api/packages/bdrelling/instruments.fyi/badge?type=platforms
https://swiftpackageindex.com/api/packages/bdrelling/instruments.fyi/badge?type=swift-versions
```

A combination of those two would be the only way to do this right now. Also, please let us know when we get the build results wrong as we'd like to investigate those failures.

**Reply by bdrelling** [reply]
Understood, and thank you for the quick reply! I'm glad (and also could have assumed) that you're aware of this, I don't think it's dire in any way. Absolute worst case, I can always spoof the shields by making my own on `shields.io`, so it's not like this is a problem without numerous workarounds.

That said, I think monitoring the endpoint works great, that's more than acceptable!

If I end up writing a GitHub Actions Workflow for this (assuming I can find time to work on it), I'll be sure to make it reusable for monitoring any repo and share with you -- could be a good candidate for the "managing your package" documentation piece.

Thanks again, Dave!
